pumps/sql_aggregate.go

```diff
@@ -147,7 +160,10 @@ func (c *SQLAggregatePump) WriteData(ctx context.Context, data []interface{}) er
 				}
 				rec.ProcessStatusCodes()
 
-				tx := c.db.WithContext(ctx).Create(rec)
+				tx := c.db.WithContext(ctx).Clauses(clause.OnConflict{
+					Columns:   []clause.Column{{Name: "id"}},
+					DoUpdates: clause.Assignments(rec.GetAssignments(table)),
```

Comment:
Can you show an example of the resulting query? (you can find it in postgres logs)

Reply:
@buger the query changes depending on what fields have values for the aggregation but this is one example:
```
INSERT INTO "tyk_aggregated_20210706" ("id","counter_hits","counter_success","counter_error","counter_request_time","counter_total_request_time","counter_identifier","counter_human_identifier","counter_last_time","counter_open_connections","counter_closed_connections","counter_bytes_in","counter_bytes_out","counter_max_upstream_latency","counter_min_upstream_latency","counter_total_upstream_latency","counter_upstream_latency","counter_max_latency","counter_min_latency","counter_total_latency","counter_latency","counter_error_map","counter_error_list","timestamp","org_id","dimension","dimension_value","code_1x","code_200","code_201","code_2x","code_301","code_302","code_303","code_304","code_3x","code_400","code_401","code_403","code_404","code_429","code_4x","code_500","code_501","code_502","code_503","code_504","code_5x") VALUES ('162561960060e484a75c92eb0a2192bc09apiidb5bcb53c0e36404358378369a229c61d',1,1,0,2.000000,2.000000,'b5bcb53c0e36404358378369a229c61d','uptime','2021-07-06 22:23:18.616',0,0,0,0,2,2,2,0.000000,2,2,2,0.000000,'{}','[]',1625619600,'60e484a75c92eb0a2192bc09','apiid','b5bcb53c0e36404358378369a229c61d',0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0) ON 
CONFLICT ("id") DO UPDATE SET 
"counter_hits"=tyk_aggregated_20210706.counter_hits + 1,
"counter_max_latency"=(0.5 * ((tyk_aggregated_20210706.counter_max_latency + 2) + ABS(tyk_aggregated_20210706.counter_max_latency - 2)),
"counter_max_upstream_latency"=(0.5 * ((tyk_aggregated_20210706.counter_max_upstream_latency + 2) + ABS(tyk_aggregated_20210706.counter_max_upstream_latency - 2)),
"counter_min_latency"=(0.5 * ((tyk_aggregated_20210706.counter_min_latency + 2) - ABS(tyk_aggregated_20210706.counter_min_latency - 2)) ,
"counter_min_upstream_latency"=(0.5 * ((tyk_aggregated_20210706.counter_min_upstream_latency + 2) - ABS(tyk_aggregated_20210706.counter_min_upstream_latency - 2)) ,
"counter_request_time"=((tyk_aggregated_20210706.counter_request_time * tyk_aggregated_20210706.counter_hits) + 2)/( tyk_aggregated_20210706.counter_hits +1),
"counter_success"=tyk_aggregated_20210706.counter_success + 1,"counter_total_request_time"=tyk_aggregated_20210706.counter_total_request_time + 2,
"counter_total_upstream_latency"=tyk_aggregated_20210706.counter_total_upstream_latency + 2
```